lib/services/course.ts

```diff
@@ -0,0 +1,45 @@
+import type { Prisma } from "@prisma/client";
+import { prisma } from "@lib/services/db";
+
+export type CourseWithBooks = Prisma.PromiseReturnType<typeof getCourseWithBooks>;
+
+export async function getCourseWithBooks(id: string)  {
```

Comment:
Actually, instead of adding a JSDoc/ type annotation , I think you could just update the system design document to show that this method can return `null`, like I did in https://github.com/calebmech/Book-Bazar/pull/64/files#diff-8aabee5a836b3dd6e3918a989562f7786416dd6c5e1c59e88b4b977fc773ab42R326

Reply:
Oh I wasn't sure what you meant by annotation, I will do that and update the spec